### PR TITLE
Add defined type for config file

### PR DESCRIPTION
**Pull request by call**
This PR encapsulates the following changes:

- Use [module-level Hiera data](https://puppet.com/docs/puppet/4.10/hiera_layers.html#the-module-layer) for [class parameter defaults](https://puppet.com/docs/puppet/latest/lang_classes.html#setting-class-parameter-defaults-with-hiera-data)
- Add `buildkite_agent::config` [defined type](https://puppet.com/docs/puppet/latest/lang_defined_types.html) to manage config files and all available [settings](https://buildkite.com/docs/agent/v3/configuration#configuration-settings).
- Format integer, string and boolean types according to Buildkite Agent Configuration [documentation](https://buildkite.com/docs/agent/v3/configuration). The config file format is:
  - [INI file](https://en.wikipedia.org/wiki/INI_file)-like
  - No [sections](https://en.wikipedia.org/wiki/INI_file#Sections)
  - double-quoted strings, unquoted booleans and integers
- Use `create_resources()` with a hash of settings hashes from Hiera (`buildkite_agent::configs`) to create multiple config files

Currently there is sample Hiera data used for testing in `data/`. The settings file created from the current sample Hiera data looks like this (from [commit b639859 action run](https://github.com/call/puppet-buildkite_agent/runs/428995035)):

```
debug=true
name="foofoobar"
priority=1
tags="queue=foobarbar"
```

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+# @summary The main class for buildkite_agent
 #
-# A description of what this class does
+# Entrypoint for the module
 #
 # @example
 #   include buildkite_agent
-class buildkite_agent {
+class buildkite_agent (
+  Optional[Hash[String, Hash[String, Variant[String, Integer, Boolean]]]] $configs,
```
Comment (dquills) [inline manifests/init.pp:8]: @call Can you advise why these config items are nested as such?

Comment (call) [reply manifests/init.pp:8]: Hey @dquills, that's a very precise [data type definition](https://puppet.com/docs/puppet/5.5/lang_data_type.html) in Puppet. The `buildkite_agent::configs` data structure must be a hash of hashes, and keys can be `str`, `bool`, or `int`. You can find an example of this data structure in the sample Hiera data, specifically this part:

```
buildkite_agent::configs:
  my-bk-config:
    config_file_path: '/usr/local/etc/buildkite-agent/buildkite-agent.cfg'
    bk_name: 'foofoobar'
    tags: 'queue=foobarbar'
    debug: true
    priority: 1
```

Multiple Buildkite agents can run on one machine, so you can use yaml data like the sample below to manage multiple config files:

```
buildkite_agent::configs:
  my-bk-config:
    config_file_path: '/usr/local/etc/buildkite-agent/buildkite-agent.cfg'
    bk_name: 'foofoobar'
    tags: 'queue=foobarbar'
    debug: true
    priority: 1
  my-bk-config-2:
    config_file_path: '/Users/call/.buildkite-agent/buildkite-agent.cfg'
    bk_name: 'myagent'
    tags: 'queue=myqueue'
    debug: false
    priority: 2
```

Comment (dquills) [reply manifests/init.pp:8]: Ah, that makes sense. Thanks for the explanation.

**Review by dquills — APPROVED** [review]
Everything looks good to me.

**Pull request merged**